src/main/java/com/apperian/api/RequestDetails.java

```diff
@@ -0,0 +1,75 @@
+package com.apperian.api;
+
+import java.io.File;
+import java.util.Map;
+
+public class RequestDetails {
+
+    // By default GET
+    private RequestMethod method = RequestMethod.GET;
+
+    private String path = null;
+
+    private Map<String, Object> data = null;
+
+    private String fileField = null;
+
+    private File file = null;
+
+    private RequestDetails() {}
+
+    public static class Builder {
+
```

Comment:
I think the RequestDetails object can be eliminated by moving the Builder class onto the ApiClient directly.  Then, the ApiClient can save the method/path/data/file on itself, without the need create and pass in a RequestDetails object.

Reply:
👍 done